src/v/cloud_storage/topic_manifest_downloader.h

```diff
@@ -57,6 +58,19 @@ class topic_manifest_downloader {
       model::timestamp_clock::duration backoff,
       topic_manifest*);
 
+    // Attempts to scan the topic manifest root paths for any topic manifests
+    // that match the given filter.
+    using tp_ns_filter_t = std::function<bool(const model::topic_namespace&)>;
+    static ss::future<result<find_topic_manifest_outcome, error_outcome>>
+    find_manifests(
+      remote& remote,
+      cloud_storage_clients::bucket_name bucket,
+      retry_chain_node& parent_retry,
+      ss::lowres_clock::time_point deadline,
+      model::timestamp_clock::duration backoff,
+      std::optional<tp_ns_filter_t> tp_filter,
```

Comment:
nit: std::function already behaves sort of like an optional std::function::bool returns false if it doesn't contain a callable function.

Reply:
Ah good to know. I'll keep this as is for now though, given it's somewhat easier to write a filter to _include_ things, and the default value should be to include everything